lib/Sema/CSRanking.cpp

```diff
@@ -939,6 +939,26 @@ SolutionCompareResult ConstraintSystem::compareSolutions(
       }
     }
 
+    // If both are properties and one is in a protocol, prefer the one not in
+    // a protocol.
+    //
+    // This is a Swift 4.1 compatibility hack. Changes elsewhere in the compiler
```

Comment:
Should we conditionalize it on language version?

Reply:
Arguably this is in line with preferring concrete implementations over protocol implementations. It is a little funny when the concrete implementation is a superclass, but…